File: frontend/src/dto/PostDto.ts
import type { CommunityDto } from "./CommunityDto";

export interface PostDto {
  id: number;
  communityDto: CommunityDto | null;
  parentPostId: number | null;
  title: string | null;
  content: string;
  createdAt: Date;
  profileId: string;
  editedAt: Date | null;
  postMediaPathList: string[];
  profilePicturePath: string;
  profileName: string;
  childrenPostsAmount: number;
  likesAmount: number;
}

File: backend/src/main/java/com/cakmak/mondatelier/dto/PostDto.java
package com.cakmak.mondatelier.dto;

import java.util.Date;
import java.util.List;

public record PostDto(
        Long id,
        Long communityId,
        String profileId,
        Long parentPostId,
        String title,
        String content,
        Date createdAt,
        Date editedAt,
        List<String> postMediaPathList
) {
}
